sdk/core/Azure.Core/src/RequestUriBuilder.cs

```diff
@@ -116,7 +117,11 @@ public string Path
             }
         }
 
-        private bool HasQuery => _queryIndex != -1;
+        /// <summary> Gets whether or not this instance of <see cref="RequestUriBuilder"/> has path. </summary>
+        public bool HasPath => _pathAndQuery.Length > 0;
```

Comment:
Shouldn't this be something like:
```suggestion
        public bool HasPath => _pathAndQuery.Length > QueryLength;
```

Reply:
I've missed that there are `PathLength` and `QueryLength` properties that are more useful, so I'll make those protected instead.